src/black/__init__.py

```diff
@@ -3425,13 +3429,7 @@ def do_transform(self, line: Line, string_idx: int) -> Iterator[TResult[Line]]:
 
         new_line = line.clone()
         new_line.comments = line.comments.copy()
-        try:
-            append_leaves(new_line, line, LL[: string_idx - 1])
-        except BracketMatchError:
-            # HACK: I believe there is currently a bug somewhere in
-            # right_hand_split() that is causing brackets to not be tracked
-            # properly by a shared BracketTracker.
```

Comment:
NB: This bug still remains, and surfaces when the lines https://github.com/psf/black/pull/1958/files#diff-d23a6917d7262ca9815093bffb7148299e516575251959ed9b405b648e76aa29R5383-R5384 are removed.